stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -142,15 +150,33 @@ internal class PaymentOptionsActivity : BaseSheetActivity<PaymentOptionResult>()
                 }
             }
         }
+    }
 
-        viewModel.userSelection.observe(this) { paymentSelection ->
-            if (paymentSelection != null) {
-                onActionCompleted(paymentSelection)
-            }
+    @VisibleForTesting
+    private val viewStateObserver = { viewState: ViewState.PaymentOptions? ->
+        val addButton = viewBinding.addButton
+        when (viewState) {
+            is ViewState.PaymentOptions.Ready -> addButton.updateState(
+                state = PrimaryButton.State.Ready(addButtonLabel)
+            )
+            is ViewState.PaymentOptions.StartProcessing -> addButton.updateState(
+                state = PrimaryButton.State.Confirming
+            )
+            is ViewState.PaymentOptions.FinishProcessing -> addButton.updateState(
+                state = PrimaryButton.State.Completed,
+                completeCallback = {
+                    onActionCompleted(viewState.result)
```

Comment:
It feels more consistent to transition to `ViewState.PaymentOptions.Finished` here instead of directly closing the sheet by calling `onActionCompleted`. Then we could always signal sheet close with the same ViewState.

Reply:
got it